src/components/application_manager/src/commands/mobile/system_request.cc

```diff
@@ -53,12 +56,360 @@ uint32_t SystemRequest::index = 0;
+          LOG4CXX_WARN(logger_,
+                       error_prefix_ << "Can't find URL scheme in json file.");
+          result = false;
+        } else {
+          if (app_data[os_type][json::urlScheme].asString().length() >
+              max_url_scheme_lengh_) {
+            LOG4CXX_WARN(
+                logger_,
+                error_prefix_
+                    << "An urlscheme length exceeds maximum allowed ["
+                    << app_data[os_type][json::urlScheme].asString().length()
+                    << "]>[" << max_url_scheme_lengh_ << "]");
+            result = false;
+          }
+        }
+      }
+      if (os_type.empty()) {
+        if (app_data.keyExists(json::android)) {
+          os_type = json::android;
+          if (!app_data[os_type].keyExists(json::packageName)) {
+            LOG4CXX_WARN(logger_,
+                         error_prefix_
+                             << "Can't find package name in json file.");
+            result = false;
+          } else {
+            if (app_data[json::android][json::packageName].asString().length() >
+                max_package_name_length_) {
+              LOG4CXX_WARN(logger_,
+                           error_prefix_
+                               << "Package name length ["
+                               << app_data[json::android][json::packageName]
+                                      .asString()
+                                      .length() << "] exceeds max length ["
+                               << max_package_name_length_ << "]in json file.");
+              result = false;
+            }
+          }
+        }
+      }
+
+      if (os_type.empty()) {
+        LOG4CXX_WARN(logger_,
+                     error_prefix_
+                         << "Can't find mobile OS type in json file.");
+        result = false;
+        continue;
+      }
+
+      // Languages verification
+      if (!app_data[os_type].keyExists(json::languages)) {
+        LOG4CXX_WARN(logger_, error_prefix_ << "\"languages\" not exists");
+        result = false;
+        continue;
+      }
+      if (!ValidateLanguages(app_data[os_type][json::languages]))
+        result = false;
+    }
+    return result;
+  }
+
+  bool ValidateAppIdAndAppName(
+      const smart_objects::SmartObject& app_data) const {
+    // Verify appid
+    if (!app_data.keyExists(json::appId)) {
+      LOG4CXX_WARN(logger_, error_prefix_ << "Can't find app ID in json file.");
+      return false;
+    }
+    bool result = true;
+    // Verify appid lenght
+    const std::string app_id(app_data[json::appId].asString());
+    if (app_id.length() > max_app_id_length_) {
+      LOG4CXX_WARN(logger_,
+                   error_prefix_
+                       << "An Object ID length exceeds maximum allowed ["
+                       << app_id.length() << "]>[" << max_app_id_length_
+                       << "]");
+      result = false;
+    }
+
+    // Verify that appid is unique
```

Comment:
@okoreniak, as you mentioned below the `set` is more suitable. So I propose to use it.
Also I propose to pass `app_ids_map_` as method param. And create it before the loop on the caller side. There is only one place where this method is used, so I don't see the point the private member. Also this will make redundant the call of clear. 
By the way, rename it. Give more meaningful name.E.g. `validated_app_ids` or `processed_app_ids` or something better. The name should make the comment on line 217 redundant(so it will be removed).